src/manifest.js

```diff
@@ -0,0 +1,140 @@
+     * - YYYY-MM-DD string: the closest match (may be an exact match)
+     * 
+     * In the future this will be expanded to parse other kinds of
+     * dateDescriptors (e.g. a descriptor meaning "6 months ago").
+     * 
+     * When there is not an exact match we interpret closest to be
+     * the closest but more recent version.
+     */
+    closestDate(dateDescriptor) {
+      if (!dateDescriptor.match(/^\d{4}-\d{2}-\d{2}$/)) {
+        return false;
+      }
+      const availableDates = (this.data?.versions || []).map((v) => v.date);
+      if (!availableDates.length) {
+        return false;
+      }
+      let closest = availableDates[0];
+      if (closest < dateDescriptor) {
+        return LATEST;
+      }
+      for (const date of availableDates) {
+        if (date >= dateDescriptor) {
+          closest = date;
+        } else {
+          break;
+        }
+      }
+      return closest;
+    }
+
+
+    /**
+     * Given a YYYY-MM-DD string, return the available subresources and their corresponding URLs
+     */
+    subresourceUrls(date) {
+      for (const version of this.data?.versions || []) {
+        if (version.date===date) {
+          return version.fileUrls;
+        }
+      }
+      throw new Error(`Attempted to access a specific date which doesn't exist in the manifest: ${date}`)
+    }
+
+  }
+
+  return {ResourceVersions}
+
+})();
+
+
+
+/**
+ * Updates provided data in-place by fetching the file on disk.
+ * TODO - either fetch from some (configurable) endpoint, or turn it into
+ * a push update by exposing an API
+ * TODO - we could do partial updates, but for the release MVP it's ok
+ * to update the entire structure
+ * @param {*} data 
+ */
+async function update(data) {
+  let newResources;
+  if (process.env.LOCAL_MANIFEST) {
+    utils.verbose(`Updating available resources via ${process.env.LOCAL_MANIFEST}`);
+    newResources = JSON.parse(
+      await (new Promise((resolve, reject) => {
+        fs.readFile(process.env.LOCAL_MANIFEST, (err, data) => {
+          /* Don't catch any error here as this is only used for development and
+          we should exit loudly */
+          if (err) reject(err);
+          resolve(data);
+        })
+      })).then((contents) =>
+        process.env.LOCAL_MANIFEST.endsWith('.gz') ? gunzip(contents) : contents
+      )
+    )
+  } else {
+    const [Bucket, Key] = ['nextstrain-inventories', 'resources.json.gz'];
+    utils.verbose(`Updating available resources by fetching s3://${Bucket}/${Key}`);
+    const S3 = new AWS.S3();
+    try {
+      newResources = JSON.parse(
+        await (new Promise((resolve, reject) => {
+          S3.getObject({Bucket, Key}, (err, data) => {
+            if (err) reject(err);
+            /* NOTE - the Body property of the response is a Buffer in S3 SDK v2
+            (which is the current version in the nextstrain.org codebase) but
+            this has changed to a readable stream in SDK v3 */
+            resolve(data.Body)
```

Comment:
Uncaught error when running with an IAM user without the required permissions to read this object
```
Error [TypeError]: Cannot read properties of null (reading 'Body')
```

Reply:
I added policy changes in bf8981831c4bd5e90461742e8b1a9d1370312c07.

Reply:
Thanks! For clarity, the underlying IAM error was fixed by me adding permissions to read the key `resources.json.gz` which is why the review app is now working, however I still need to change the code so that errors while getting/parsing the object are handled appropriately.

Reply:
Fixed via force-push (error was because reject was falling through, and a thrown error in an async callback isn't captured by the surrounding promise)